File: backend/credit-wizard-api/credit-wizard-api/Dtos/SemesterPlannerDto.cs
using credit_wizard_api.Models;
using System.ComponentModel.DataAnnotations;

namespace credit_wizard_api.Dtos
{
    public class SemesterPlannerDto
    {
        public Guid Id { get; set; }

        [Required]
        public Guid UserId { get; set; }

        public User? User { get; set; }

        public bool Completed { get; set; }

        [Required]
        public Guid SemesterId { get; set; }
        public SemesterDto? SemesterDto { get; set; }

        public List<SemesterPlannerModulDto> SemesterPlannerModulDtos { get; set; } = new List<SemesterPlannerModulDto>();

        [Required]
        public Guid SemesterTimeSlotId { get; set; }

        public SemesterTimeSlotDto? SemesterTimeSlotDto { get; set; }
    }
}

File: frontend/credit-wizard/src/app/shared/dtos/semesterPlannerDto.ts
import { Guid } from "guid-typescript";
import { IUserDto } from "./userDto";
import { ISemesterDto } from "./semesterDto";
import { ISemesterPlannerModulDto } from "./semesterPlannerModulDto";
import { ISemesterTimeSlotDto } from "./semesterTimeSlotDto";

export interface ISemesterPlannnerDto{
    id: Guid,
    userId: Guid,
    user: IUserDto,
    completed: boolean,
    semesterId: boolean,
    semesterDto: ISemesterDto,
    semesterPlannerModulDtos: ISemesterPlannerModulDto[],
    semesterTimeSlotId: Guid,
    semesterTimeSlotDto: ISemesterTimeSlotDto
}